db/crud.go

```diff
@@ -870,6 +870,30 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+func (db *DatabaseCollectionWithUser) computeHLVLogic(d *Document) (*Document, error) {
+	newVVEntry := CurrentVersionVector{}
+	bucketUUID, err := db.dbCtx.Bucket.UUID()
+	if err != nil {
+		return nil, err
+	}
+	newVVEntry.SourceID = bucketUUID
+	// add cas here this will be macro expanded
+	// need to keep same value for now or adding new version vector entry will fail with new cas < than old cas
```

Comment:
This feels error-prone - like it would be easy to introduce a bug where we either fail to update the value later, or accidentally macro expand over a version that been provided by CBL that we should be preserving.  Alternatives might be to use VersionCAS=MaxUint64 as "should macro expand", or add a flag to HybridLogicalVector that has the same meaning.

Reply:
I have changed this temporarily to MaxUint64 before adding the extra logic here. I think i'm a bit confused on the logic i'm missing here. Am I right on saying if we have a CBL mutation, they update the source version pair accordingly, we then need to preserve that right? And instead we macro expand the cvCAS on the HLV? Then the same for ISGR?